### PR TITLE
add support for KNN, AdaBoostStump and RandomForest

### LINKED ISSUE #70: Weitere ML-Modelle
- [ ] AdaBoostStump
- [ ] RandomForest
- [ ] KNN
- [ ] Correlation stage NA Values fixen
- [ ] Baseline Model (No data manipulation after segmentation)

### PR DESCRIPTION
**Pull request by yvokeller**

### CODE REVIEW

**Comment by gabrieltorresgamez** [comment]
should i discontinue this pull request because of #75 ?

**Comment by yvokeller** [comment]
I suggest we merge this one, my plan is to then create "model/\*" type  branches from "main" for each ML model type and then tune hyperparams there. These "model/\*" branches will then probably never be merged back to main.

**Comment by ghost** [comment]
# RUN: AdaBoostStumpClassifier, 2023-05-17 07-41-38

## Hyperparameters

| Path        | Param                                  | main                           | workspace               |
|-------------|----------------------------------------|--------------------------------|-------------------------|
| params.yaml | dvclive.model                          | HistGradientBoostingClassifier | AdaBoostStumpClassifier |
| params.yaml | dvclive.model_hparams.learning_rate    | 0.2                            | 0.5                     |
| params.yaml | dvclive.model_hparams.max_bins         | 255                            | -                       |
| params.yaml | dvclive.model_hparams.max_iter         | 2000                           | -                       |
| params.yaml | dvclive.model_hparams.max_leaf_nodes   | 31                             | -                       |
| params.yaml | dvclive.model_hparams.min_samples_leaf | 20                             | -                       |
| params.yaml | dvclive.model_hparams.n_estimators     | -                              | 200                     |
| params.yaml | dvclive.model_hparams.random_state     | -                              | 42                      |
| params.yaml | dvclive.seed                           | 41                             | 42                      |
| params.yaml | train_test_split_ratio02.type          | person                         | measurement             |

## Metrics

| Path                 | Metric          | main    | workspace   | Change   |
|----------------------|-----------------|---------|-------------|----------|
| dvclive/metrics.json | test.epoch.acc  | 0.39211 | 0.65738     | 0.26526  |
| dvclive/metrics.json | test.epoch.f1   | 0.31129 | 0.64935     | 0.33806  |
| dvclive/metrics.json | train.epoch.acc | 1.0     | 0.82947     | -0.17053 |
| dvclive/metrics.json | train.epoch.f1  | 1.0     | 0.81224     | -0.18776 |

## Plots

### Confusion Matrix

Confusion Matrix

**Review by ghost — APPROVED** [review]
LGTM -Gabo

**Review by gabrieltorresgamez — APPROVED** [review]

**Comment by ghost** [comment]
# RUN: AdaBoostStumpClassifier, 2023-05-19 16-15-12

## Hyperparameters

| Path   | Param   | main   | workspace   |
|--------|---------|--------|-------------|

## Metrics

| Path   | Metric   | main   | workspace   | Change   |
|--------|----------|--------|-------------|----------|

## Plots

### Confusion Matrix

Confusion Matrix

**Comment by ghost** [comment]
# RUN: AdaBoostStumpClassifier, 2023-05-19 16-20-26

## Hyperparameters

| Path   | Param   | main   | workspace   |
|--------|---------|--------|-------------|

## Metrics

| Path   | Metric   | main   | workspace   | Change   |
|--------|----------|--------|-------------|----------|

## Plots

### Confusion Matrix

Confusion Matrix

**Comment by ghost** [comment]
# RUN: AdaBoostStumpClassifier, 2023-05-19 16-24-36

## Hyperparameters

| Path   | Param   | main   | workspace   |
|--------|---------|--------|-------------|

## Metrics

| Path   | Metric   | main   | workspace   | Change   |
|--------|----------|--------|-------------|----------|

## Plots

### Confusion Matrix

Confusion Matrix
